### ISSUE TITLE
Create --clean flag for containerized environments

### ISSUE THREAD

**Issue by timreimherr** [transferwise/pipelinewise#573]
### Subject
Pipelinewise fails to run in containerized environments due to leftover pid files from previous executions. When using pipelinewise in containerized environments you need to have volumes to persist data from the `import` process which is then used in the `run_tap` process. However, leftover pid files are saved in the volume from the `import` process which then causes pipelinewise to think that a process is running, then it logs the message `logger_name=pipelinewise.cli.pipelinewise log_level=INFO message=Tap Salesforce is currently running` and the container dies.

Could you add a `--clean` flag to remove pid files before a process completes?

### Your environment
* Version 0.26.0
* tap-saleforce
* target-redshift

### Steps to reproduce
Create a Docker image that contains:
- Pipelinewise installed
- configured tap and target files
- a script to run commands to the installed pipelinewise
- set up a persistent volume to store tap/target config data from the `import` process

### Expected behaviour
The data import is successful.

### Actual behaviour
We get the message `logger_name=pipelinewise.cli.pipelinewise log_level=INFO message=Tap Salesforce is currently running` and the container dies.

**Comment by koszti** [comment]
Pid files are deleted automatically on successful and on failure runs and when the taps stopped by `SIGINT` and `SIGTERM`. Pids should remain only if the tap stopped by `SIGKILL` (kill -9). Do you know how and what stopped the running taps?

How would the `--cleanup` option work? Would that going through all directories in `~/.pipelinewise` and deleting every pid files? My only concern with the cleanup option then if a tap is really running then it can remove the valid pid files as well and will be possible to start the same tap multiple times, which should never happen.

How would you avoid to not running the same tap in parallel, if we add `--cleanup` option?

**Comment by EamonKeane** [comment]
@timreimherr did you ever come upon a solution for this?

It happens rarely enough that I haven't added a solution, but I think it happens when a pre-emptible node is killed by GKE.

> Due to the nature of GKE and preemptible nodes, once a node is scheduled for deletion, a node receives a SIGTERM, but the underlying pod never knows it’s going to die until it is actually terminated. Again, for stateless services, this causes no concern since GKE simply spins up new nodes which the pods can be scheduled on.

My thinking is to wrap the pipelinewise execution in a script which checks for the present of the specific files associated with that tap/target combination, and if it finds it sleeps for 2* the typical execution duration (10 mins). Then it deletes the files and continues running.

This is likely to cause issues only if:
* The node wasn't actually killed and is still running
* You are attempting to execute another task duing a long running operation such as copying a large table - care is needed around this if you're executing pipelinewise frequently.

**Comment by EamonKeane** [comment]
Also just to mention, this issue may be resolved with this recent patch to make kubelet handle node shutdown gracefully.

https://github.com/kubernetes/kubernetes/pull/96129

**Comment by iljau** [comment]
> Pid files are deleted automatically on successful and on failure runs and when the taps stopped by `SIGINT` and `SIGTERM`. Pids should remain only if the tap stopped by `SIGKILL` (kill -9).

My observation is that on SIGTERM log files are cleaned up (`'Stopping gracefully...'` appears in logs), but .pid file remains.
